bindings/go/blob/filesystem/filesystem.go

```diff
@@ -0,0 +1,130 @@
+package filesystem
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+var ErrReadOnly = fmt.Errorf("read only file system")
+
+// FileSystem is an interface that needs to be fulfilled by any filesystem implementation
+// to be usable within the OCM Bindings.
+// The ComponentVersionReference Implementation is the osFileSystem which is backed by the os package.
+type FileSystem interface {
+	Base() string
+	Open(name string) (fs.File, error)
+	OpenFile(name string, flag int, perm os.FileMode) (fs.File, error)
+	MkdirAll(name string, perm os.FileMode) error
+	Remove(name string) error
+	ReadDir(name string) ([]fs.DirEntry, error)
+	RemoveAll(path string) error
+	Stat(name string) (fs.FileInfo, error)
+	ReadOnly() bool
+	ForceReadOnly()
+}
+
+// File is an interface that needs to be fulfilled by any file implementation
+// to be usable within the OCM Bindings.
+// The File is a typical file implementation that is also writeable.
+type File interface {
+	fs.File
+	io.Writer
+}
+
+func NewFS(base string, flag int) (FileSystem, error) {
+	base, err := filepath.Abs(base)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get absolute path: %w", err)
+	}
+	fi, err := os.Stat(base)
+	if os.IsNotExist(err) {
+		if flag&os.O_CREATE == 0 {
+			return nil, fmt.Errorf("path does not exist: %s", base)
+		}
+		if err = os.MkdirAll(base, 0o755); err != nil {
+			return nil, fmt.Errorf("failed to create path: %w", err)
+		}
+	} else if err != nil {
+		return nil, fmt.Errorf("unable to stat path: %w", err)
+	}
+	// fi might be nil if we just create the directory in the os.IsNotExist
+	// branch
+	if fi != nil && !fi.IsDir() {
+		return nil, fmt.Errorf("path is not a directory: %s", base)
+	}
+	return &osFileSystem{base: base, flag: flag}, nil
+}
+
+type osFileSystem struct {
+	base string
+
+	flagMu sync.RWMutex
+	flag   int
```

Comment:
I would expect some comment what kind of flag is meant here. From reading the code, I got the impression that `flag` represents conditions.

Reply:
I think I can rename to mode to make it clear that this is about the filesystem mode. However its not only mode because its a bitmask so it can also contain other attributes.

Reply:
A comment what `flag` should contains would be enough for me

Reply:
Added a comment with a reference to OpenFile which is where I got the name of the parameter from